### ISSUE TITLE
Sway 1.8-dev crash on window close

### ISSUE THREAD

**Issue by hissssst** [swaywm/sway#7303]
- **Sway Version:**
  `sway version 1.8-dev`

- **Debug Log:**
  [sway2.log](https://github.com/swaywm/sway/files/10135158/sway2.log)

- **Configuration File:**
https://0x0.st/ok-4.txt

- **Stack Trace:**
No stack trace, just `(core dumped) exited with code 1`

- **Description:**
This bug rarely happens, but when I close firefox window (`Mozilla Firefox 102.5.0esr`), sway crashes

**Comment by bl4ckb0ne** [comment]
```
11:36:33.556 [DEBUG] [wlr] [backend/drm/renderer.c:315] Failed to get DMA-BUF from buffer
11:36:33.556 [DEBUG] [wlr] [backend/drm/drm.c:423] connector eDP-1: Failed to import buffer for scan-out
```

Strange.

Do you run firefox with the wayland backend?

**Comment by hissssst** [comment]
> Do you run firefox with the wayland backend?

Yes

**Comment by hissssst** [comment]
> Are you able to provide a stacktrace?

How? The process just finishes up without any stacktrace and shell shows `(core dumped) exited with code 1`

**Comment by bl4ckb0ne** [comment]
I don't know what distro you're using, coredumpctl, gdb trace, anything that can point to a problem in the code. We can't do much with a "that bug rarely happens" and a sway log that has no significant error.

**Comment by hissssst** [comment]
[sway.coredump.txt](https://github.com/swaywm/sway/files/10144920/sway.coredump.txt)

Here you go

**Comment by hissssst** [comment]
But I think this is a coredump from a latest crash. However, it is caused by the same actions (closing firefox)

**Comment by hissssst** [comment]
@emersion , what does "waiting on reporter" label mean? Do I need to provide more information?

**Comment by hissssst** [comment]
@bl4ckbone I've found a way to reproduce it
1. Open any window
2. Move it to scratchpad
3. Hide it
4. Kill the window process

**Comment by hissssst** [comment]
Maybe next week. If you were able to reproduce the issue and this commit fixes it, just close it

**Comment by bl4ckb0ne** [comment]
@hissssst i made some changes to the patch. If you could test it, it would be greatly appreciated.

**Comment by emersion** [comment]
Can you confirm this is fixed in master?

**Comment by Zeioth** [comment]
I just tested master and I got the same

    11:36:33.556 [DEBUG] [wlr] [backend/drm/renderer.c:315] Failed to get DMA-BUF from buffer
    11:36:33.556 [DEBUG] [wlr] [backend/drm/drm.c:423] connector eDP-1: Failed to import buffer for scan-out

message every 100ms on the log

**Comment by eternal-sorrow** [comment]
Almost reliably reproduces every time I close VSCode window.

**Comment by bl4ckb0ne** [comment]
@eternal-sorrow care to share some logs?

**Comment by eternal-sorrow** [comment]
Nope, didn't get a useful stacktrace even with debug symbos. My stacktrace looks like this:
```
#0  0x00007f30c69677cc in ?? ()
#1  0x00007ffcf099c010 in ?? ()
#2  0x7afbd8b7b329cb00 in ?? ()
#3  0x0000000000000006 in ?? ()
#4  0x00007f30c5cfae00 in ?? ()
#5  0x00007ffcf099c020 in ?? ()
#6  0x00007f30b4014000 in ?? ()
#7  0x0000000000000001 in ?? ()
#8  0x00007f30c6917292 in ?? ()
#9  0x00007f30c6aadcb0 in ?? ()
#10 0x00007f30c6901472 in ?? ()
#11 0x0000000000000020 in ?? ()
#12 0x0000001c00000780 in ?? ()
#13 0x0000000000000000 in ?? ()
```

**Comment by eternal-sorrow** [comment]
For some reason I can't reproduce the crash after I started sway with debug output.

**Comment by bl4ckb0ne** [comment]
@hissssst can you confirm your issue is fixed in master?

**Comment by eternal-sorrow** [comment]
Okay, I set up constant debug logging for sway in case I will be able to reproduce the crash, and yes, it happened again. There is nothing unusual in the log though:
```
82:03:27.201 [INFO] [sway/commands.c:260] Handling command 'kill'
82:03:27.279 [DEBUG] [sway/ipc-server.c:328] Sending window::close event
82:03:27.280 [DEBUG] [sway/tree/arrange.c:263] Usable area for ws: 1920x1052@0,0
82:03:27.280 [DEBUG] [sway/tree/arrange.c:293] Arranging workspace '5' at 0.000000, 0.000000
82:03:27.280 [DEBUG] [sway/desktop/transaction.c:398] Transaction 0x55774f544a70 committing with 2 instructions
82:03:27.280 [DEBUG] [sway/desktop/transaction.c:294] Applying transaction 0x55774f544a70
```
And this is it. 'kill` command is me closing VSCode window.

**Comment by bl4ckb0ne** [comment]
What's unusual?

**Comment by eternal-sorrow** [comment]
Nothing. That's what I've said. But these were the last lines in the log before sway crashed.

**Comment by hissssst** [comment]
Fixed in master

**Comment by eternal-sorrow** [comment]
Will there be a hotfix release where this is fixed?

**Comment by emersion** [comment]
Sway 1.8.1 is planned, yes.

**Comment by eternal-sorrow** [comment]
Crashes when closing VSCode still occur on 1.8.1. I guess, it's a different bug.

**Comment by bl4ckb0ne** [comment]
Open a new issue with the stacktrace please

**Comment by eternal-sorrow** [comment]
I already wrote before that I was unable to get a meaningful stacktrace.
